pyomo/contrib/parmest/parmest.py

```diff
@@ -508,40 +968,32 @@ def _Q_opt(
 
         # Solve the extensive form with ipopt
         if solver == "ef_ipopt":
-            if not calc_cov:
-                # Do not calculate the reduced hessian
-
-                solver = SolverFactory('ipopt')
-                if self.solver_options is not None:
-                    for key in self.solver_options:
-                        solver.options[key] = self.solver_options[key]
-
-                solve_result = solver.solve(self.ef_instance, tee=self.tee)
-
             # The import error will be raised when we attempt to use
             # inv_reduced_hessian_barrier below.
             #
             # elif not asl_available:
             #    raise ImportError("parmest requires ASL to calculate the "
             #                      "covariance matrix with solver 'ipopt'")
-            else:
-                # parmest makes the fitted parameters stage 1 variables
-                ind_vars = []
-                for ndname, Var, solval in ef_nonants(ef):
-                    ind_vars.append(Var)
-                # calculate the reduced hessian
-                (solve_result, inv_red_hes) = (
-                    inverse_reduced_hessian.inv_reduced_hessian_barrier(
-                        self.ef_instance,
-                        independent_variables=ind_vars,
-                        solver_options=self.solver_options,
-                        tee=self.tee,
-                    )
+
+            # parmest makes the fitted parameters stage 1 variables
+            ind_vars = []
+            for ndname, Var, solval in ef_nonants(ef):
```

Comment:
`ndname` and `solval` should probably be `nd_name` and `sol_val`. Also I'm not sure what `ef_nonants` is supposed to mean?